lib/WebService/Async/SmartyStreets.pm

```diff
@@ -0,0 +1,183 @@
+                );
+            IO::Async::Loop->new->add($ss);
+            return $ss;
+        }
+    }
+    
+    my $ss = get_smartystreets();
+
+    my $addr = $ss->verify_international(<hash of address element>, geocode => 'true')->get;
+    return $addr->status;
+    
+=head1 DESCRIPTION
+
+This class calls the SmartyStreets API and parse the response to `WebService::Async::SmartyStreets::Address`
+
+This module uses Future::AsyncAwait, visit https://metacpan.org/pod/Future::AsyncAwait for more information.
+
+=over 4
+
+=cut
+
+use parent qw(IO::Async::Notifier);
+
+use mro;
+no indirect;
+
+use URI;
+use URI::QueryParam;
+
+use Future::AsyncAwait;
+use Net::Async::HTTP;
+use JSON::MaybeUTF8 qw(:v1);
+
+use WebService::Async::SmartyStreets::Address;
+
+use Log::Any qw($log);
+
+=head2 configure
+
+Configures the class with the auth_id and token
+Takes in: Hash which consists of auth_id and token
+
+=cut
+
+sub configure {
+    my ($self, %args) = @_;
+    for my $k (qw(auth_id token)) {
+        $self->{$k} = delete $args{$k} if exists $args{$k};
+    }
+    $self->next::method(%args);
+}
+
+sub auth_id { shift->{auth_id} }
+sub token   { shift->{token} }
+
+sub next_id {
+    ++(shift->{id} //= 'AA00000000');
+}
+
+=head2 ua
+
+Creates a User agent (Net::Async::HTTP) that is used to make connection to the SmartyStreets API
+
+=cut
+
+sub ua {
+    my ($self) = @_;
+    $self->{ua} //= do {
+        $self->add_child(
+            my $ua = Net::Async::HTTP->new(
+                fail_on_error            => 1,
+                decode_content           => 1,
+                pipeline                 => 0,
+                max_connections_per_host => 4,
+                user_agent =>
+                    'Mozilla/4.0 (WebService::Async::SmartyStreets; BINARY@cpan.org; https://metacpan.org/pod/WebService::Async::SmartyStreets)',
+            ));
+        $ua;
+        }
+}
+
+=head2 verify_international, verify_usa
+
+Calls to different API depending on the country of the address.
+Takes in: Hash of address and required components
+
+Components includes:
+    - country (compulsory)
+    - address1
+    - address2
+    - organization	
+    - locality
+    - administrative_area
+    - postal_code
+    - geocode (true or false)
+    
+Returns: WebService::Async::SmartyStreets::Address object
+
+=cut
+
+async sub verify_international {
+    my ($self, %args) = @_;
+    my $uri = URI->new('https://international-street.api.smartystreets.com/verify');
+    return await $self->verify($uri => %args);
+}
+async sub verify_usa {
+    my ($self, %args) = @_;
+    my $uri = URI->new('https://us-street.api.smartystreets.com/street-address');
+    return await $self->verify($uri => %args);
+}
+
+
+=head2 verify
+
+Makes connection to SmartyStreets API and parses the response into WebService::Async::SmartyStreets::Address.
+Takes in: URI object and hash of address args
+Returns: WebService::Async::SmartyStreets::Address object
+
+=cut
+
+async sub verify {
```

Comment:
Would this be better as a private sub instead, given that the public APIs are `verify_international` and `verify_usa`?

Reply:
alright will change